### PR TITLE
Optimizations in bn254 ECMUL

### PR DESCRIPTION
**Pull request by garyschulte**
Optimizations for ecmul in gnark-crypto eip196 imple for ECMUL:

* use .Double() when ecmul when scalar == 2 
* if point is infinity encoding, just return infinity

### CODE REVIEW

**Review by pinges — APPROVED** [review]
LGTM

**Comment by garyschulte** [comment]
waiting on fuzz testing results before merging

File: gnark/gnark-jni/gnark-eip-196.go [hunk 1]
```diff
@@ -128,6 +127,11 @@ func eip196altbn128G1Mul(javaInputBuf, javaOutputBuf, javaErrorBuf *C.char, cInp
     // Convert input C pointers to Go slice
     input := (*[EIP196PreallocateForG1 + EIP196PreallocateForScalar]byte)(unsafe.Pointer(javaInputBuf))[:inputLen:inputLen]
 
+    // inifinity check:
+    if isAllZeroEIP196(input, 0, 64) {
+        return 0
```
Comment (guidovranken) [inline gnark/gnark-jni/gnark-eip-196.go:132]: Add `*outputLen = EIP196PreallocateForG1`

File: gnark/gnark-jni/gnark-eip-196.go [hunk 2]
```diff
@@ -128,6 +127,11 @@ func eip196altbn128G1Mul(javaInputBuf, javaOutputBuf, javaErrorBuf *C.char, cInp
     // Convert input C pointers to Go slice
     input := (*[EIP196PreallocateForG1 + EIP196PreallocateForScalar]byte)(unsafe.Pointer(javaInputBuf))[:inputLen:inputLen]
 
+    // inifinity check:
```
Comment (guidovranken) [inline gnark/gnark-jni/gnark-eip-196.go:130]: typo

**Pull request merged**